spatial-lib/src/cmr/spatial/circle.clj

```diff
@@ -0,0 +1,106 @@
+(ns cmr.spatial.circle
+  (:require
+   [cmr.spatial.math :as math :refer :all]
+   [primitive-math]
+   [cmr.spatial.point :as p]
+   [cmr.spatial.derived :as d]
+   [cmr.spatial.geodetic-ring :as gr]
+   [cmr.spatial.validation :as sv]
+   [cmr.spatial.polygon :as poly]
+   [cmr.common.dev.record-pretty-printer :as record-pretty-printer])
+  (:import
+   cmr.spatial.geodetic_ring.GeodeticRing
+   cmr.spatial.polygon.Polygon
+   cmr.spatial.point.Point))
+
+(primitive-math/use-primitive-operators)
+
+;; Circle
+(defrecord Circle
+  [
+   ^Point center
+   ^double radius
+   ])
+
+(record-pretty-printer/enable-record-pretty-printing Circle)
+
+(defn circle
+  "Creates a new minimum bounding rectangle"
+  ([^Point center ^double radius]
+   (->Circle center radius))
+  ([^double lon ^double lat ^double radius]
+   (->Circle (p/point lon lat) radius)))
+
+(def MIN_RADIUS
+  "Minimum radius in meters"
+  10)
+
+(def MAX_RADIUS
+  "Maximum radius in meters"
+  1000000)
+
+(def ^:const ^double EARTH_RADIUS_APPROX
+  "Radius of the earth in meters for polygon approximation of the circle"
+  6378137.0)
+
+(defn- validate-center
```

Comment:
What would happen if a vertex point landed on a pole? Is this a possibility?

Reply:
I don't think it is possible for math to get there which is infinity.

Reply:
Why can't the center be on a pole? And is it ok for the center to be near a pole?

Reply:
Near the pole is OK. On the pole causes the math calculation to be exactly zero and we can't draw the circle at all.